config/config.go

```diff
@@ -65,12 +68,16 @@ type Config struct {
 	Extend      Extend
 	Path        string
 	Description string
-	Rules       map[string]Rule
-	Allowlist   Allowlist
+	Rules       map[string]rule.Rule
+	Allowlist   rule.Allowlist
 	Keywords    map[string]struct{}
 
 	// used to keep sarif results consistent
 	OrderedRules []string
+
+	// report-related settings.
+	ReportPath string
+	Reporter   report.Reporter
```

Comment:
@rgmz was this the line causing the circular dependency? Couldn't the reporter be added to the `Detector` struct like any other cli flag? Or hell, better yet, keep reporting compare logic in the `cmd` package. I'm not a fan of moving `allowlist.go` into a `rule` package since allowlists can be globally set as well.

Reply:
I believe the issue was that one of the reporters expected a reference to Config. Maybe it'd work under detector — I'll check.

Reply:
Okay, I've moved the reporter to the `Detector` struct.